### PR TITLE
Add readthedocs.io Documentation for End Users

### PR DESCRIPTION
**Pull request by alex-mckenna**
In order to satisfy #951, it seems a good idea to bring in [this documentation effort](https://github.com/thoughtpolice/clash-docs) to the main project, and update it to

  * include the additional documentation around flags we want
  * refer to the most recent stable version of clash

As this documentation was not created by us, the repository has been merged in to preserve history (and provenance of commits). The PR will also be on hold as draft until we get confirmation that @thoughtpolice is okay with the documentation being merged into the main repo, and maintained by the `clash-lang` maintainers.

### CODE REVIEW

**Comment by thoughtpolice** [comment]
I can add someone as a maintainer to the clash-lang project on readthedocs, so if a member of the org can sign up and give me their readthedocs username, I'll add it. (I'm hesitant to delete it and re-use it in case RtD blocks re-used names, I'm not sure if it supports that.)

I think all you'll need to do is just update the Git repository to point to this copy (I think), and maybe look over the branching/build settings a bit.

**Comment by martijnbastiaan** [comment]
@thoughtpolice Cool! I've made an account: `hmbastiaan`.

**Comment by thoughtpolice** [comment]
@martijnbastiaan I've added you, if you have any questions feel free to ask!

**Comment by martijnbastiaan** [comment]
@thoughtpolice Thanks! I think all the relevant options are there. I'll experiment a bit on a separate account before changing the pointers to this repo.

@alex-mckenna Could you redo this PR without the flags commit? I've written up a good way of merging remote projects on the [clash-cosim merge PR](https://github.com/clash-lang/clash-compiler/pull/652).

**Comment by alex-mckenna** [comment]
Rebased to get rid of the unrelated clash flags commits. What's the status on the stuff you were looking at for this issue @martijnbastiaan?

**Comment by martijnbastiaan** [comment]
@alex-mckenna Haven't had the time to look at it yet. I think we can simply merge this PR and figure out how to go from there though!

**Comment by alex-mckenna** [comment]
Whoops, must've got distracted by something shiny when doing that rebase. I'll just fix that...

File: docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/_build
```
Comment (martijnbastiaan) [inline docs/.gitignore]: Move this do `/.gitignore`

File: README.md [hunk 2]
```diff
@@ -115,3 +115,4 @@ Or [use Nix](https://nixos.org/nix/download.html) to get a shell with the `clash
 ```bash
 nix-shell
 ```
+
```
Comment (martijnbastiaan) [inline README.md]: empty change

File: docs/LICENSE [hunk 3]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2012-2017 University of Twente
```
Comment (martijnbastiaan) [inline docs/LICENSE]: We can omit this file. It's the same license as `/LICENSE`.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Needed as an awkward work around to
```
Comment (martijnbastiaan) [inline requirements.txt]: move to `docs/`

File: examples/MAC.hs [hunk 5]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (martijnbastiaan) [inline examples/MAC.hs]: :thinking:

File: examples/FIR.hs [hunk 6]
```diff
@@ -38,3 +39,13 @@ testBench = done
     done           = expectedOutput (topEntity clk rst (enableGen) testInput)
     clk            = tbSystemClockGen (not <$> done)
     rst            = systemResetGen
+=======
```
Comment (martijnbastiaan) [inline examples/FIR.hs]: :thinking:

**Pull request merged**